packages/client/manifest.json

```diff
@@ -5618,29 +5617,37 @@
             "label": "Clicked row",
             "key": "row"
           }
-        ],
-        "dependsOn": {
-          "setting": "allowEditRows",
-          "value": false
-        }
+        ]
       },
       {
         "type": "boolean",
         "label": "Add rows",
         "key": "allowAddRows",
-        "defaultValue": true
+        "defaultValue": true,
+        "dependsOn": {
+          "setting": "datasource.type",
+          "value": ["table", "viewV2"]
+        }
       },
       {
         "type": "boolean",
         "label": "Edit rows",
         "key": "allowEditRows",
-        "defaultValue": true
+        "defaultValue": true,
+        "dependsOn": {
+          "setting": "datasource.type",
+          "value": ["table", "viewV2"]
+        }
```

Comment:
When testing, this locked row editing for non `["table", "viewV2"]` grids into `true` with no way to disable it. If they cannot disable editing, their onRowClick behaviour will never be accessible. In these scenarios we may need to ensure that editing is resolved to false in the grid component itself

Reply:
Yep you're correct that this locks it at whatever the current setting is, but that's how all those conditional settings behave. We just need to ensure that we properly handle that inside the components, and we do in this case. If you use a non DS+ inside the grid then the prop is ignored, because it will override certain features based on the datasource type.

So even though the prop will be true, you'll notice that you can't actually edit rows.